@xen-orchestra/lite/src/libs/vm.ts

```diff
@@ -0,0 +1,26 @@
+import { saveAs } from "file-saver";
+import { useVmStore } from "@/stores/vm.store";
+import type { XenApiVm } from "@/libs/xen-api";
+
+export function exportVmsAsJsonFile(
+  vmRefs: XenApiVm["$ref"][],
+  fileName: string
+) {
+  const { getByOpaqueRef: getVm } = useVmStore().subscribe();
```

Comment:
An external function should not subscribe to a store.

It would be preferable to let the `VmActionExportItem` component to subscribe, then pass the VMs to the function.

So instead of accepting a `vmRefs` argument, the function should accept a `vms: (XenApiVm | undefined)[]` argument